processing/src/main/java/org/apache/druid/segment/IndexIO.java

```diff
@@ -663,9 +650,32 @@ public QueryableIndex load(File inDir, ObjectMapper mapper, boolean lazy, Segmen
           segmentBitmapSerdeFactory.getBitmapFactory(),
           columns,
           smooshedFiles,
-          metadata,
           lazy
-      );
+      )
+      {
+        @Override
+        public Metadata getMetadata()
+        {
+          try {
+            ByteBuffer metadataBB = smooshedFiles.mapFile("metadata.drd");
```

Comment:
I doesn't really know these parts; but this will mean that the `Metadata` will be read from disk every time its asked for; is that desired?

Reply:
Not necessarily.  The smooshedFiles is memory mapped and this is returning a memoized memory-mapped buffer.  It will do some less than fast operations, but not necessarily brand new disk IO.  The getMetadata() call is used seldom (only a few specific introspection-oriented calls like the segmentMetadata query) so it's better to have it be lazy and not consume on-heap memory.